src/widgets/kernel/qtooltip.cpp

```diff
@@ -475,6 +534,82 @@ void QToolTip::showText(const QPoint &pos, const QString &text, QWidget *w, cons
     }
 }
 
+/*!
+    Shows \a layout as a tool tip, with the global position \a pos as
+    the point of interest. The tool tip will be shown with a platform
+    specific offset from this point of interest.
+
+    If you specify a non-empty rect the tip will be hidden as soon
+    as you move your cursor out of this area.
+
+    The \a rect is in the coordinates of the widget you specify with
+    \a w. If the \a rect is not empty you must specify a widget.
+    Otherwise this argument can be \nullptr but it is used to
+    determine the appropriate screen on multi-head systems.
+
+    The \a msecDisplayTime parameter specifies for how long the tool tip
+    will be displayed, in milliseconds. With the default value of -1, the
+    time is based on the count of the layout.
+
+    If \a layout is empty the tool tip is hidden. If the layout is the
+    same as the currently shown tooltip, the tip will \e not move.
+    You can force moving by first hiding the tip with an empty layout,
+    and then showing the new tip at the new position.
+*/
+
+void QToolTip::showLayout(const QPoint& pos, QLayout* layout, QWidget* w, const QRect& rect, int msecDisplayTime)
+{
+    if (!layout)
+        return;
+    if (QTipLabel::instance && QTipLabel::instance->isVisible()) { // a tip does already exist
+        if (!layout->count()){ // empty layout means hide current tip
+            QTipLabel::instance->hideTip();
+            return;
+        } else if (!QTipLabel::instance->fadingOut) {
+            // If the tip has changed, reuse the one
+            // that is showing (removes flickering)
+            QPoint localPos = pos;
+            if (w)
+                localPos = w->mapFromGlobal(pos);
+            if (QTipLabel::instance->tipChanged(localPos, layout, w)){
+                QTipLabel::instance->reuseTip(layout, msecDisplayTime, pos);
+                QTipLabel::instance->setTipRect(w, rect);
+                QTipLabel::instance->placeTip(pos, w);
+            }
+            return;
+        }
+    }
+
+    if (layout->count()) { // no tip can be reused, create new tip:
+        QWidget *tipLabelParent = [w]() -> QWidget* {
+#ifdef Q_OS_WIN32
```

Comment:
Are we expected to cater to any OS that is not Windows? This will only be used by LF right?

Reply:
It is. The intention is to keep it as similar to showText as possible, so that when we update QT version it will be easier to update showLayout if showText will be changed in the next QT version.